infrastructure/src/main/java/org/corfudb/infrastructure/BatchProcessor.java

```diff
@@ -114,127 +102,152 @@ private void recordRunnable(Runnable fsyncRunnable, Optional<Timer> fsyncTimer)
-                } else if (streamLog.quotaExceeded() && currOp.getMsg().getPriorityLevel() != PriorityLevel.HIGH) {
-                    currOp.getFutureResult().completeExceptionally(
-                            new QuotaExceededException("Quota of "
-                                    + streamLog.quotaLimitInBytes() + " bytes"));
-                    log.warn("batchprocessor: quota exceeded, dropping msg {}", currOp.getMsg());
-                } else if (currOp.getType() == Type.SEAL && currOp.getMsg().getEpoch() >= sealEpoch) {
-                    log.info("batchWriteProcessor: updating from {} to {}", sealEpoch, currOp.getMsg().getEpoch());
-                    sealEpoch = currOp.getMsg().getEpoch();
-                    res.add(currOp);
-                    processed++;
-                    lastOp = currOp;
-                } else if (currOp.getMsg().getEpoch() != sealEpoch) {
-                    log.warn("batchWriteProcessor: wrong epoch on {} msg, seal epoch is {}, and msg epoch is {}",
-                            currOp.getType(), sealEpoch, currOp.getMsg().getEpoch());
-                    currOp.getFutureResult().completeExceptionally(new WrongEpochException(sealEpoch));
-                    res.add(currOp);
-                    processed++;
-                    lastOp = currOp;
+                } else if (streamLog.quotaExceeded() &&
+                        (currentOp.getRequest().getHeader().getPriority() != PriorityLevel.HIGH)) {
+                    currentOp.getFutureResult().completeExceptionally(
+                            new QuotaExceededException("Quota of " + streamLog.quotaLimitInBytes() + " bytes"));
+
+                    log.warn("batchWriteProcessor: quota exceeded, dropping request {}",
+                            TextFormat.shortDebugString(currentOp.getRequest()));
+                } else if (currentOp.getType() == BatchWriterOperation.Type.SEAL &&
+                        (currentOp.getRequest().getPayload().getSealRequest().getEpoch() >= sealEpoch)) {
+                    log.info("batchWriteProcessor: updating epoch from {} to {}",
+                            sealEpoch, currentOp.getRequest().getPayload().getSealRequest().getEpoch());
+
+                    sealEpoch = currentOp.getRequest().getPayload().getSealRequest().getEpoch();
+                    res.add(currentOp);
+                    numProcessed++;
+                    lastOp = currentOp;
+                } else if (currentOp.getRequest().getHeader().getEpoch() != sealEpoch) {
+                    log.warn("batchWriteProcessor: wrong epoch on {} request, seal epoch is {}, and request epoch is {}",
+                            currentOp.getType(), sealEpoch, currentOp.getRequest().getHeader().getEpoch());
+
+                    currentOp.getFutureResult().completeExceptionally(new WrongEpochException(sealEpoch));
+                    res.add(currentOp);
+                    numProcessed++;
+                    lastOp = currentOp;
```

Comment:
The point of batch writing is to amortize the cost of the fsync, but in this case the request is rejected and doesn't produce any io and thus shouldn't contribute to the batch size.

Reply:
Done.